config_src/infra/FMS1/MOM_coms_infra.F90

```diff
@@ -438,6 +439,36 @@ subroutine min_across_PEs_real_1d(field, length, pelist)
   call mpp_min(field, length, pelist)
 end subroutine min_across_PEs_real_1d
 
+!> Implementation of any() intrinsic across PEs
+function any_across_PEs(field, pelist)
+  logical, intent(in)           :: field      !< Local PE value
+  integer, optional, intent(in) :: pelist(:)  !< List of PEs to work with
+  logical :: any_across_PEs
+
+  integer :: field_flag
+
+  ! FMS1 does not support logical collectives, so integer flags are used.
+  field_flag = 0
+  if (field) field_flag = 1
+  call max_across_PEs(field_flag, pelist)
+  any_across_PEs = (field_flag > 0)
+end function any_across_PEs
+
+!> Implementation of all() intrinsic across PEs
+function all_across_PEs(field, pelist)
+  logical, intent(in)           :: field      !< Local PE value
+  integer, optional, intent(in) :: pelist(:)  !< List of PEs to work with
+  logical :: all_across_PEs
+
+  integer :: field_flag
+
+  ! FMS1 does not support logical collectives, so integer flags are used.
+  field_flag = 0
+  if (field) field_flag = 1
+  call min_across_PEs(field_flag, pelist)
+  all_across_PEs = (field_flag < 1)
```

Comment:
As written, the logic on lines 468 and 469 seems to be implementing the parallel version of `.not.any()`, and not the parallel version of `all()`.  To correctly implement `all_across_PEs()` as the analog of `all()`, I think that line should be `all_across_PEs = (field_flag > 0)`.

Reply:
Looks like you are right.  As you might have guessed, `all_across_PEs` was not used anywhere.  Perhaps adding it was a mistake.

I believe this is better position in the `infra` layer, because the conversions to integers and use of `{min,max}_across_PEs` reflects the absence of logical MPI collectives in FMS and is thus a uniquely FMS operation.  A different framework would not have this restriction.

Reply:
`all_across_PEs` test has been updated